diff --git a/src/iceberg/avro/avro_data_util.cc b/src/iceberg/avro/avro_data_util.cc
index 8c2b6f0..2f09b5f 100644
--- a/src/iceberg/avro/avro_data_util.cc
+++ b/src/iceberg/avro/avro_data_util.cc
@@ -22,7 +22,6 @@
 #include <arrow/array/builder_decimal.h>
 #include <arrow/array/builder_nested.h>
 #include <arrow/array/builder_primitive.h>
 #include <arrow/extension_type.h>
-#include <arrow/json/from_string.h>
 #include <arrow/type.h>
 #include <arrow/util/decimal.h>
 #include <avro/Generic.hh>
diff --git a/src/iceberg/parquet/parquet_reader.cc b/src/iceberg/parquet/parquet_reader.cc
index 51f602b..f1ff631 100644
--- a/src/iceberg/parquet/parquet_reader.cc
+++ b/src/iceberg/parquet/parquet_reader.cc
@@ -123,11 +123,12 @@ class ParquetReader::Impl {
     // Open the Parquet file reader
     ICEBERG_ASSIGN_OR_RAISE(input_stream_, OpenInputStream(options));
     auto file_reader =
         ::parquet::ParquetFileReader::Open(input_stream_, reader_properties);
-    ICEBERG_ARROW_ASSIGN_OR_RETURN(
-        reader_, ::parquet::arrow::FileReader::Make(pool_, std::move(file_reader),
-                                                    arrow_reader_properties));
+    std::unique_ptr<::parquet::arrow::FileReader> reader;
+    ICEBERG_ARROW_RETURN_NOT_OK(::parquet::arrow::FileReader::Make(
+        pool_, std::move(file_reader), arrow_reader_properties, &reader));
+    reader_ = std::move(reader);

     // Project read schema onto the Parquet file schema
     ICEBERG_ASSIGN_OR_RAISE(projection_, BuildProjection(reader_.get(), *read_schema_));
     metadata_context_ = {.file_path = options.path,
@@ -240,13 +241,14 @@ class ParquetReader::Impl {

     // Create the record batch reader
     if (row_group_indices.empty()) {
       // None of the row groups are selected, return an empty record batch reader
       context_->record_batch_reader_ = std::make_unique<EmptyRecordBatchReader>();
     } else {
       auto column_indices = SelectedColumnIndices(projection_);
-      ICEBERG_ARROW_ASSIGN_OR_RETURN(
-          context_->record_batch_reader_,
-          reader_->GetRecordBatchReader(row_group_indices, column_indices));
+      std::unique_ptr<::arrow::RecordBatchReader> record_batch_reader;
+      ICEBERG_ARROW_RETURN_NOT_OK(reader_->GetRecordBatchReader(
+          row_group_indices, column_indices, &record_batch_reader));
+      context_->record_batch_reader_ = std::move(record_batch_reader);
     }

     return {};
